runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropPlugin.java

```diff
@@ -85,20 +91,38 @@ public void onWidgetHidden(WidgetHiddenChanged event)
 		}
 
 		String text = widget.getText();
+		final Stream<Widget> siblingWidgets = Arrays.stream(widget.getParent().getDynamicChildren());
+		final List<Integer> spriteIDs = siblingWidgets.map(Widget::getSpriteId).collect(Collectors.toList());
+
 		if (text != null)
 		{
+			int color = widget.getTextColor();
+
 			switch (prayer)
 			{
 				case MELEE:
-					widget.setTextColor(config.getMeleePrayerColor().getRGB());
+					if (spriteIDs.contains(SpriteID.SKILL_ATTACK)
```

Comment:
Leave previous variable as stream and use .filter(predicate).findAny().isPresent()